### ISSUE TITLE
Crash if gnome-keyring isn't installed (using kde for example)

### ISSUE THREAD

**Issue by marco44** [borgbase/vorta#235]
Using vorta 0.6.13 (from archlinux/AUR) under KDE.
```
2019-03-24 08:58:23,280 - vorta.borg.borg_thread - DEBUG - Using VortaSecretStorageKeyring keyring to store passwords.
2019-03-24 08:58:23,281 - asyncio - DEBUG - Using selector: EpollSelector
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/secretstorage/util.py", line 47, in send_and_get_reply
    return self._connection.send_and_get_reply(msg)
  File "/usr/lib/python3.7/site-packages/jeepney/integrate/blocking.py", line 49, in send_and_get_reply
    return future.result()
jeepney.wrappers.DBusErrorResponse: [org.freedesktop.DBus.Error.ServiceUnknown] ('The name org.freedesktop.secrets was not provided by any .service files',)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/vorta/views/repo_add_dialog.py", line 128, in run
    params = BorgInfoThread.prepare(self.values)
  File "/usr/lib/python3.7/site-packages/vorta/borg/info.py", line 28, in prepare
    ret = super().prepare(profile)
  File "/usr/lib/python3.7/site-packages/vorta/borg/borg_thread.py", line 114, in prepare
    ret['password'] = keyring.get_password('vorta-repo', profile.repo.url)
  File "/usr/lib/python3.7/site-packages/vorta/keyring/secretstorage.py", line 25, in get_password
    collection = secretstorage.get_default_collection(self.connection)
  File "/usr/lib/python3.7/site-packages/secretstorage/collection.py", line 161, in get_default_collection
    return Collection(connection)
  File "/usr/lib/python3.7/site-packages/secretstorage/collection.py", line 44, in __init__
    self._collection.get_property('Label')
  File "/usr/lib/python3.7/site-packages/secretstorage/util.py", line 65, in get_property
    (signature, value), = self.send_and_get_reply(msg)
  File "/usr/lib/python3.7/site-packages/secretstorage/util.py", line 56, in send_and_get_reply
    raise SecretServiceNotAvailableException(data) from resp
secretstorage.exceptions.SecretServiceNotAvailableException: The name org.freedesktop.secrets was not provided by any .service files
KCrash: crashing... crashRecursionCounter = 2
KCrash: Application Name = python3.7 path = /usr/bin pid = 29665
KCrash: Arguments: /usr/bin/python3.7 -f 
KCrash: Attempting to start /usr/lib/drkonqi from kdeinit
sock_file=/run/user/1000/kdeinit5__0
```

Adding gnome-keyring solves the issue. But I'd rather have an error message than a crash :) . Is gnome-keyring really a dependency ?

**Comment by m3nu** [comment]
Hi, thanks for reporting. We recently changed the password implementation for Linux and it still needs some fine-tuning to cover all possible setups.

As immediate fix, I'll check for this error when the app starts and fall back to saving the password to the database.

As next step, we could implement a KDE-specific password storage, if it exists.

**Comment by marco44** [comment]
Yes, there's one, kwallet. Very similar to gnome-keyring...

**Comment by mitya57** [comment]
Why not use the [Python Keyring Lib](https://github.com/jaraco/keyring) which already has both SecretStorage and KWallet backends?

**Comment by m3nu** [comment]
Cool. Will look into it – except if you want to add support for it? 😏

Anyways, I've added [some more functions](https://github.com/m3nu/vorta/commit/df7c92e76ee0708e6aab40ecc55c2981d74b2fab) to the init-code. We should now catch this error on startup and fall back to saving passwords in the database.

Here a debug package you can try:

```
$ pip uninstall vorta
$ pip install https://files.qmax.us/vorta/vorta-0.6.13-debug.tar.gz
```

After this if should work and you should see something like `Using VortaDBKeyring Keyring implementation.`

**Comment by m3nu** [comment]
> Why not use the [Python Keyring Lib](https://github.com/jaraco/keyring) which already has both SecretStorage and KWallet backends?

We used this for a while, but it caused too many issues. E.g. not possible to package it when using the macOS keyring. On Linux it also gave errors during startup. So we switched to a lighter library.

**Comment by mitya57** [comment]
> We used this for a while, but it caused too many issues. E.g. not possible to package it when using the macOS keyring. On Linux it also gave errors during startup. So we switched to a lighter library.

Ok, understood.

**Comment by m3nu** [comment]
Is the debug-package/PR working for you? If yes, I'll release it as (another) hotfix right away.

`pip install https://files.qmax.us/vorta/vorta-0.6.13-debug.tar.gz`

**Comment by marco44** [comment]
Yes. This solves the problem. Great !

**Comment by m3nu** [comment]
Great. Released as [v0.6.14](https://github.com/borgbase/vorta/releases/tag/v0.6.14).

If there is anything else we can improve for Linux desktops and saving passwords, let me know.
